src/Blueprint/Blueprint/Cmdlets/NewAzureRMBlueprintAssignment.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.Azure.Commands.Blueprint.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Blueprint.Models;
+using System;
+using System.Collections;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Language;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Properties;
+using ParameterSetNames = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterSetNames;
+using ParameterHelpMessages = Microsoft.Azure.Commands.Blueprint.Common.BlueprintConstants.ParameterHelpMessages;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.Commands.Blueprint.Common;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "BlueprintAssignment", SupportsShouldProcess = true, DefaultParameterSetName = ParameterSetNames.CreateBlueprintAssignment), OutputType(typeof(PSBlueprintAssignment))]
+    public class NewAzureRmBlueprintAssignment : BlueprintCmdletBase
+    {
+        #region Parameters
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.BlueprintAssignmentName)]
+        [ValidatePattern("^[0-9a-zA-Z_-]*$", Options = RegexOptions.Compiled | RegexOptions.CultureInvariant)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.BlueprintObject)]
+        [ValidateNotNull]
+        public PSBlueprintBase Blueprint { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.SubscriptionIdToAssign)]
+        [ValidateNotNullOrEmpty]
+        public string[] SubscriptionId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.Location)]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.Resources/resourceGroups")]
```

Comment:
@filizt usually the `LocationCompleter` determines the valid locations for the resource being created -- is a Blueprint assignment a tracked resource? If so, we should make sure the argument for this completer accurately reflects the correct resource type.

Reply:
We chatted about this offline. Keeping it as is for now.

---
In reply to: [261402747](https://github.com/Azure/azure-powershell/pull/8537#discussion_r261402747) [](ancestors = 261402747)